lib/cors-anywhere.js

```diff
@@ -342,6 +344,19 @@ function getHandler(options, proxy) {
       return;
     }
 
+    var target = location.protocol + '//' + location.hostname;
+    if (corsAnywhere.targetBlacklist.indexOf(target) >= 0) {
+      res.writeHead(403, 'Forbidden', cors_headers);
+      res.end('The target "' + target + '" was blacklisted by the operator of this proxy.');
+      return;
+    }
+
+    if (corsAnywhere.targetWhitelist.length && corsAnywhere.targetWhitelist.indexOf(target) === -1) {
+      res.writeHead(403, 'Forbidden', cors_headers);
+      res.end('The target "' + target + '" was not whitelisted by the operator of this proxy.');
+      return;
+    }
+
```

Comment:
This check can be bypassed if the whitelisted target site has an open redirect. A better place to enforce this is in the `proxyRequest` function.

Locally I started with something like this (many months ago), but I never got to finish the implementation.

```diff
 function proxyRequest(req, res, proxy) {
   var location = req.corsAnywhereRequestState.location;
+  // TODO: Add something like this?
+  // For https://github.com/Rob--W/cors-anywhere/issues/67
+  if (req.corsAnywhereRequestState.checkRequestAllowed &&
+      !req.corsAnywhereRequestState.checkRequestAllowed(location)) {
+      res.writeHead(403, 'Forbidden', withCORS({}, req));
+      res.end('The requested resource was blocked by the operator of this proxy.');
+    return;
+  }
```

The idea behind this is that the validator can be as flexible as anyone wants to, with some default implementation in `server.js` that simply looks at the environment variables and the prefix (like you're doing right now).

Matching by prefix is the simplest, but is it sufficient for most use cases? I recall another bug that wanted to match by "file extension".